thoth-api/src/errors.rs

```diff
@@ -29,6 +29,11 @@ pub enum ThothError {
     EntityNotFound,
     #[fail(display = "Issue's Work and Series cannot have different Imprints.")]
     IssueImprintsError,
+    #[fail(
+        display = "{} is not a validly formatted {} and will not be saved",
+        _0, _1
+    )]
+    IdentifierParseError(String, String),
```

Comment:
Thinking about errors in [the other discussion](https://github.com/thoth-pub/thoth/pull/249#discussion_r654310803), it probably makes sense to have an error variant per identifier (e.g. `DoiParseError`, `OrcidParseError`) as the first input `String` is never dynamically generated (e.g. always `"DOI".to_string()`